Lock based bug when we hammer the server

```
$ docker run -p 9000:9000 scalawilliam/eventsource-hub
$ echo $(seq 1 200) > sample-file.txt
$ ab -k -n 20000000 -c 100 -p sample-file.txt http://localhost:9000/target
```

In console, get:
```
play.api.UnexpectedException: Unexpected exception[OverlappingFileLockException: null]
	at play.api.http.HttpErrorHandlerExceptions$.throwableToUsefulException(HttpErrorHandler.scala:253)
	at play.api.http.DefaultHttpErrorHandler.onServerError(HttpErrorHandler.scala:184)
	at play.core.server.AkkaHttpServer$$anonfun$$nestedInanonfun$executeHandler$1$1.applyOrElse(AkkaHttpServer.scala:199)
	at play.core.server.AkkaHttpServer$$anonfun$$nestedInanonfun$executeHandler$1$1.applyOrElse(AkkaHttpServer.scala:198)
	at scala.concurrent.Future.$anonfun$recoverWith$1(Future.scala:412)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:37)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:60)
	at play.api.libs.streams.Execution$trampoline$.executeScheduled(Execution.scala:109)
	at play.api.libs.streams.Execution$trampoline$.execute(Execution.scala:71)
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:68)
Caused by: java.nio.channels.OverlappingFileLockException: null
	at sun.nio.ch.SharedFileLockTable.checkList(FileLockTable.java:255)
	at sun.nio.ch.SharedFileLockTable.add(FileLockTable.java:152)
	at sun.nio.ch.FileChannelImpl.lock(FileChannelImpl.java:1063)
	at java.nio.channels.FileChannel.lock(FileChannel.java:1053)
	at services.FileStore.appendLine(FileStore.scala:47)
	at services.ChannelStore$AtChannel.$anonfun$push$3(ChannelStore.scala:52)
	at services.ChannelStore$AtChannel.$anonfun$push$3$adapted(ChannelStore.scala:50)
	at scala.util.Success.$anonfun$map$1(Try.scala:251)
	at scala.util.Success.map(Try.scala:209)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:287)
```

--- a/app/services/ChannelStore.scala
+++ b/app/services/ChannelStore.scala
@@ -9,6 +9,7 @@ import model.ChannelId
 import play.api.libs.EventSource.Event
 import play.api.libs.iteratee.Concurrent
 import play.api.libs.iteratee.streams.IterateeStreams
+import play.api.inject.ApplicationLifecycle
 
 import scala.concurrent.{ExecutionContext, Future}
 
@@ -16,7 +17,7 @@ import scala.concurrent.{ExecutionContext, Future}
   * Created by me on 06/04/2017.
   */
 @Singleton
-class ChannelStore @Inject()()(implicit executionContext: ExecutionContext) {
+class ChannelStore @Inject()(applicationLifecycle: ApplicationLifecycle)(implicit executionContext: ExecutionContext) {
   cs =>
 
   type Channel = (FileStore, Source[Event, NotUsed], Concurrent.Channel[Event])
@@ -63,5 +64,7 @@ class ChannelStore @Inject()()(implicit executionContext: ExecutionContext) {
         case (_, source, _) => source
       }
   }
+
+  applicationLifecycle.addStopHook(() => Future.successful(channelsData.get().foreach(_._2._1.close())))
 
 }

--- a/app/services/FileStore.scala
+++ b/app/services/FileStore.scala
@@ -21,6 +21,10 @@ class FileStore(val eventsPath: Path) {
     Files.createFile(eventsPath)
   }
 
+
+  private val raf = new java.io.RandomAccessFile(eventsPath.toFile, "rw").getChannel
+  private val lock = raf.lock()
+
   /**
     * Typically used to achieve the Last-Event-ID part of the EventSource spec
     * that allows consumers to resume from where they left off in case of a failure
@@ -43,11 +47,12 @@ class FileStore(val eventsPath: Path) {
 
   /** Use approach from https://gist.github.com/ScalaWilliam/37c4ef3c41e656a6d3c02d992f5c6191 **/
   def appendLine(line: String): Unit = {
-    val raf = new java.io.RandomAccessFile(eventsPath.toFile, "rw").getChannel
-    val lock = raf.lock()
     val lineBytes = (line + "\n").getBytes("UTF-8")
-    try Files.write(eventsPath, lineBytes, StandardOpenOption.APPEND)
-    finally lock.release()
+    Files.write(eventsPath, lineBytes, StandardOpenOption.APPEND)
+  }
+
+  def close(): Unit = {
+    lock.release()
   }
 
 }

